src/cheri_insts.sail

```diff
@@ -383,53 +380,29 @@ union clause ast = CAndPerm : (regidx, regidx, regidx)
  * previous value and bits 0 to [cap_hperms_width]-1 of integer register *rs2*
  * and the **uperms** field set to the bitwise and of its previous value and
  * bits [cap_uperms_shift] to [cap_uperms_shift]+[cap_uperms_width]-1 of *rs2*.
- *
- * ## Exceptions
- *
- * An exception is raised if:
- *   - *cs1*.**tag** is not set.
- *   - *cs1* is sealed.
  */
 function clause execute(CAndPerm(cd, cs1, rs2)) = {
-  let cs1_val = C(cs1);
+  let cs1_val = clearTagIfSealed(C(cs1));
```

Comment:
Hmm, yeah, it's a shame because I didn't want to hide away the fact that it's the register value you're working on... How would you feel about `input_val` or `input_cap` for the result of maybe clearing the tag?

Reply:
Okay, I've avoided this idiom. Does this look any better?